Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py

```diff
@@ -157,25 +157,30 @@ def get_events_command(client, args):
     )
 
 
-def fetch_incidents(client, last_run, first_fetch_time, event_type_filter, threat_type, threat_status, limit=50):
+def fetch_incidents(client, last_run, first_fetch_time, event_type_filter, threat_type, threat_status, limit=50,
+                    integration_context=None):
+    incidents: list = []
+    # check if there're incidents saved in context
+    if integration_context:
+        remained_incidents = integration_context.get('incidents')
+        # return incidents if exists in context.
+        if remained_incidents:
+            return last_run, remained_incidents[:limit], remained_incidents[limit:]
     # Get the last fetch time, if exists
     last_fetch = last_run.get('last_fetch')
-
     # Handle first time fetch, fetch incidents retroactively
     if not last_fetch:
         last_fetch, _ = parse_date_range(first_fetch_time, date_format=DATE_FORMAT, utc=True)
-    incidents: list = []
     fetch_times = get_fetch_times(last_fetch)
     fetch_time_count = len(fetch_times)
     for index, fetch_time in enumerate(fetch_times):
         if index < fetch_time_count - 1:
-            raw_events = client.get_events(interval=fetch_time + "/" + fetch_times[index + 1],
-                                           event_type_filter=event_type_filter,
-                                           threat_status=threat_status, threat_type=threat_type)
+            last_fetch = fetch_times[index + 1]
         else:
-            raw_events = client.get_events(interval=fetch_time + "/" + get_now().strftime(DATE_FORMAT),
-                                           event_type_filter=event_type_filter,
-                                           threat_status=threat_status, threat_type=threat_type)
+            last_fetch = get_now().strftime(DATE_FORMAT)
+        raw_events = client.get_events(interval=fetch_time + "/" + last_fetch,
```

Comment:
rename `fetch_time` to be `start_query_time`
rename `last_fetch` to be `end_query_time`

in the `next_run = {'last_fetch': end_query_time}`